### PR TITLE
Feature decorate a ruby class

### PR DESCRIPTION
**Pull request by Beardless-sheik**
In this pull request:

- Created a classroom class
- Set relationships with the Classroom and Student class
- Created a Book and Rental class
- Create a relationship between the Person and Book using the intermediate class Rental [ that is instantiated by (date, person, book)]

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# This book class shows a Book as being an object in the Library(it hasa  title & author attribue & a rentals Array)
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (Dipeshtwis) [inline book.rb]: - [ ] Getters and Setters for @rentals should be present

` attr_accessor :rentals`

File: student.rb [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 # This Student class can be initialised by passing classroom, Age, Name & parent permission.
 class Student < Person
+  attr_accessor :classroom
+
```
Comment (Dipeshtwis) [inline student.rb]: - [ ] only getter is required for `classroom` 
- [ ] It should a method that serves as the 'belongs-to' side (a student belongs to a classroom)
```ruby

  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Beardless-sheik, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

- [ ] As per the project requirement, A person has many rentals. Please implement it.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N.A.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾🎉✔**
Hello @Beardless-sheik 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**